service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/jobs/JobClient.java

```diff
@@ -216,28 +292,32 @@ public synchronized JobResult getJob(String jobId)
     {
         URL url;
 
-        /* Codes_SRS_JOBCLIENT_21_024: [If the JobId is null, empty, or invalid, the getJob shall throws IllegalArgumentException.] */
-        if((jobId == null) || jobId.isEmpty())
+        if (Tools.isNullOrEmpty(jobId))
         {
             throw new IllegalArgumentException("null jobId");
         }
 
-        /* Codes_SRS_JOBCLIENT_21_025: [The getJob shall create a URL for Jobs using the iotHubConnectionString.] */
         try
         {
-            url = iotHubConnectionString.getUrlJobs(jobId);
+            url = IotHubConnectionString.getUrlJobs(this.hostName, jobId);
         }
         catch (MalformedURLException e)
         {
             throw new IllegalArgumentException("Invalid JobId to create url");
         }
 
-        /* Codes_SRS_JOBCLIENT_21_026: [The getJob shall send a GET request to the iothub using the created url.] */
-        /* Codes_SRS_JOBCLIENT_21_027: [If the getJob failed to send a GET request, it shall throw IOException.] */
-        /* Codes_SRS_JOBCLIENT_21_028: [If the getJob failed to verify the iothub response, it shall throw IotHubException.] */
-        HttpResponse response = DeviceOperations.request(this.iotHubConnectionString, url, HttpMethod.GET, new byte[]{}, null, USE_DEFAULT_TIMEOUT);
+        Proxy proxy = options.getProxyOptions() != null ? options.getProxyOptions().getProxy() : null;
```

Comment:
Is calling `getProxyOptions()` without capturing the output wasteful? IOW, would it be better to call it and capture the output, if tha tis null return null, otherwise call `getProxy()` off of that?

Reply:
This was mostly done to save space when looking at the code, but that's a good point. I can split this into two lines to avoid getting the options twice